fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/node/PredicateParser.java

```diff
@@ -0,0 +1,421 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query.node;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Stack;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+
+/**
+ * Basically follows Dijkstra's shunting yard algorithm to ensure
+ * correct handling of operator precedence as the expression nodes
+ * are added
+ */
+public class PredicateParser {
+    private static final Logger logger = Logger.getLogger(PredicateParser.class.getName());
+
+    // newline character for clearer errors and warnings
+    private static final String NEWLINE = System.lineSeparator();
+
+    // Stacks used to hold expression nodes until they are ready to be shunted
+    private final Stack<ExpNode> operatorStack = new Stack<>();
+    private final Stack<ExpNode> operandStack = new Stack<>();
+
+    // For parameter encoding
+    private static final String SQL_PARAM_ESCAPE_CHARACTER = "";
+    private static final String SQL_PARAM_BLACKLIST_CHARACTERS_REGEX = "['\"]";
+    private final Pattern escapeCharacterPattern;
+
+    public PredicateParser() {
+        escapeCharacterPattern = Pattern.compile(SQL_PARAM_BLACKLIST_CHARACTERS_REGEX);
+    }
+
+    /**
+     * Make sure any literal strings we're embedding in our SQL statements are safe. String parameters
+     * should almost always use bind variables (which don't require encoding).
+     * @param parameter
+     * @return
+     */
+    private final String encodeParameter(String parameter) {
+        String safeParameter = Optional.ofNullable(parameter).orElse("");
+        Matcher matcher = escapeCharacterPattern.matcher(safeParameter);
+        safeParameter = matcher.replaceAll(SQL_PARAM_ESCAPE_CHARACTER);
```

Comment:
Since we're using strings underneath shouldn't it look at `'`

Reply:
This is taken directly from the existing parameter encoding we do. It should be matching an removing any `'`.